iwyu_test_util.py

```diff
@@ -92,6 +150,43 @@ def HasTarget(target):
         supported.append(m.group(1))
     return target in supported
 
+  @staticmethod
+  def HasCPredefinedMacro(macro):
+    return macro in Features._GetCPredefinedMacros()
+
+  @staticmethod
+  def HasCXXPredefinedMacro(macro):
+    return macro in Features._GetCXXPredefinedMacros()
+  
+  @staticmethod
+  def ActiveCXXstdlib(lib):
+    id_macro = {
+      'libcxx': '_LIBCPP_VERSION',
+      'libstdcxx': '__GLIBCXX__',
+      # https://github.com/microsoft/STL/wiki/Macro-_MSVC_STL_UPDATE
+      'msvc': '_MSVC_STL_UPDATE',
+    }
+    macro = id_macro.get(lib)
+    if macro is None:
+      raise SyntaxError('Unknown C++ standard library: %s' % lib)
+
+    return macro in Features._GetCXXPreprocessorMacros()
+
+  @staticmethod
+  def ActiveCstdlib(lib):
+    id_macro = {
```

Comment:
Do you actually need the libc detection for anything upcoming? Otherwise I'm inclined to either not do it at all for now, or at least just support the particular libraries that we need (uclibc and newlib looks a little esoteric at this point).

Reply:
I was thinking the same myself.

With the test generation I'm looking at, I'm starting with C headers (to keep things simple), so probably want macos+glibc working. Newlib was actually simple to double check. If you could verify that we're correctly identifying glibc on a linux host that'd be great